src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```diff
@@ -0,0 +1,263 @@
+    // the current user's id from the list.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String userId = userService.getCurrentUser().getUserId();
+
+    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
+    // replaced with this hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    interviewers.remove(userId);
+
+    for (String interviewer : interviewers) {
+      possibleInterviews.addAll(
+          getPossibleInterviewSlotsForPerson(interviewer, range, timezoneOffset));
+    }
+
+    // We need to check that the person looking to schedule is not already scheduled during any of
+    // the proposed times.
+
+    List<ScheduledInterview> userScheduledInterviews =
+        scheduledInterviewDao.getScheduledInterviewsInRangeForUser(
+            userId, range.start(), range.end());
+
+    Set<PossibleInterviewSlot> conflictingInterviews = new HashSet<PossibleInterviewSlot>();
+
+    for (PossibleInterviewSlot slot : possibleInterviews) {
+      for (ScheduledInterview userInterview : userScheduledInterviews) {
+        if (userInterview.when().contains(Instant.parse(slot.utcEncoding()))) {
+          conflictingInterviews.add(slot);
+        }
+      }
+    }
+
+    possibleInterviews.removeAll(conflictingInterviews);
+
+    List<PossibleInterviewSlot> possibleInterviewList =
+        new ArrayList<PossibleInterviewSlot>(possibleInterviews);
+
+    sortInterviews(possibleInterviewList);
+    return possibleInterviewList;
+  }
+
+  private List<PossibleInterviewSlot> getPossibleInterviewSlotsForPerson(
+      String userId, TimeRange range, ZoneOffset timezoneOffset) {
+    List<Availability> availabilities =
+        availabilityDao.getInRangeForUser(userId, range.start(), range.end());
+    availabilities.removeIf(avail -> avail.scheduled());
+
+    List<PossibleInterviewSlot> possibleInterviewSlotsForPerson =
+        new ArrayList<PossibleInterviewSlot>();
+    for (int i = 0; i < availabilities.size() - 3; i++) {
+      if (availabilities.get(i).when().end().equals(availabilities.get(i + 1).when().start())) {
+        if (availabilities
+            .get(i + 1)
+            .when()
+            .end()
+            .equals(availabilities.get(i + 2).when().start())) {
+          if (availabilities
+              .get(i + 2)
+              .when()
+              .end()
+              .equals(availabilities.get(i + 3).when().start())) {
+            possibleInterviewSlotsForPerson.add(
+                PossibleInterviewSlot.create(
+                    availabilities.get(i).when().start().toString(),
+                    getDate(availabilities.get(i).when().start(), timezoneOffset),
+                    getTime(availabilities.get(i).when().start(), timezoneOffset)));
+          }
```

Comment:
Since `availabilities` is sorted, we can just check if the availability [i + 3] is at current time + 45 minutes

Reply:
Good call, done.